inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/SPARQLQueryStore.java

```diff
@@ -69,34 +72,87 @@ private static final String optionalLanguageFilteredValue(String aProperty, Stri
         fragment.append(" }\n");
         return fragment.toString();
     }
+
+    /**
+     * adds an OPTIONAL block which looks for a value that is declared with a
+     * subproperty of the given property
+     */
+    private static final String optionalLanguageFilteredSubPropertyValue(String aProperty,
+        String aLanguage, String aVariable)
+    {
+        StringBuilder fragment = new StringBuilder();
+        fragment.append("  OPTIONAL {\n");
+        fragment.append("    ?s ").append("?p ").append(aVariable).append(" .\n");
+        fragment.append("    ?p ").append("?pSUBPROPERTY* ").append(aProperty).append(" .\n");
+        fragment.append(languageFilterForVariable(aVariable, aLanguage));
+        fragment.append(" }\n");
+        return fragment.toString();
+    }
+  
+    /**
+     * adds an OPTIONAL block which looks for a value that is declared with a
+     * subproperty of the given property
+     */
+    private static final String optionalSubPropertyLabelFilter(Set<KBHandle> labelProperties,
```

Comment:
The method name sounds as if the returned value is the label of a sub-property. But my understanding is that the query retrieves the label taking sub-properties of the label IRI into account. Also, it doesn't seem to be a "filter" because it doesn't really filter anything. Can we not simply all it `queryForOptionalLabel()`.

Reply:
yeah, that wud be better.